taigaProject/backend/service/userstory_service.py

```diff
@@ -426,4 +428,55 @@ def get_pb_coupling(project_id, auth_token):
     except Exception as e :
         print(f"Unexpected error :{e}")
         return None
-    
\ No newline at end of file
+    
+def get_burndown_all_sprints(project_id, auth_token):
+
+
+    min_heap = []
+    end_dates = []
+
+
+    total_story_points = 0
+
+
+    date_storypoint_map = {}
+    result = {}
+    milestones_response = get_milestone_by_project_id(project_id, auth_token)
+
+    for milestone_item in milestones_response:
+
+        end_date_obj = datetime.strptime(milestone_item['estimated_finish'], "%Y-%m-%d")
+
+        heapq.heappush(min_heap, datetime.strptime(milestone_item['estimated_start'], "%Y-%m-%d"))
```

Comment:
Isn't this a bit overkill. Why not just have start_dates and get min(start_dates) for get the first date?

Reply:
I think it is more efficient than sorting a list of dates.